code/modules/overmap/exoplanets/decor/turfs/water.dm

```diff
@@ -51,3 +138,128 @@
 	name = "putrid sewage"
 	desc = "This is utterly vile."
 	color = "#9ea844"//Ew
+
+/turf/simulated/floor/exoplanet/water/shallow/sewage/process()
+	. = ..()
+	reagents.add_reagent(/singleton/reagent/toxin, 2)
+	for(var/mob/living/carbon/human/H in src)
+		// add toxins
+		H.reagents.add_reagent(/singleton/reagent/ammonia, rand(10,30))
+		H.reagents.add_reagent(/singleton/reagent/acid/hydrochloric, rand(10,30))
+		H.reagents.add_reagent(/singleton/reagent/toxin, rand(10,30))
+		for(var/A in H.organs)
+			var/obj/item/organ/external/E = A
+			if(BP_IS_ROBOTIC(E))
+				continue
+			if (E.wounds.len)
+				for(var/datum/wound/W in E.wounds)
+					if(W.germ_level < INFECTION_LEVEL_ONE)
+						W.germ_level = INFECTION_LEVEL_ONE
+					W.germ_level += rand(10,50)
+
+/turf/simulated/floor/exoplanet/water/proc/wash(atom/movable/O as obj|mob)
+
+	var/obj/effect/effect/water/W = new(O)
+	W.create_reagents(100)
+	W.reagents.add_reagent(/singleton/reagent/water, 100)
+	W.set_up(O, 100)
+
+	if(iscarbon(O))
+		var/mob/living/carbon/M = O
+		if(M.r_hand)
+			M.r_hand.clean_blood()
+		if(M.l_hand)
+			M.l_hand.clean_blood()
+		if(M.back)
+			if(M.back.clean_blood())
+				M.update_inv_back(0)
+
+		//flush away reagents on the skin
+		if(M.touching)
+			var/remove_amount = M.touching.maximum_volume * M.reagent_permeability() //take off your suit first
+			M.touching.remove_any(remove_amount)
+
+		if(ishuman(M))
+			var/mob/living/carbon/human/H = M
```

Comment:
This part of this proc is copypasted from watercloset.dm and we generally want to avoid copypasting huge chunks of code. What you should do is make this whole part a proc on /mob/living/carbon/human and then just call that both here and on the shower